release.sh

```
@@ -75,7 +75,7 @@ list_changed_submodules () {
 
 update_one_submodule () {
     local sm="$1"
-    local branch="master"
+    local branch="HEAD"
```

I doubt that's the correct way to pull the default branch. IIUC `origin/HEAD` is only a symref and not guaranteed to be updated locally (after the first clone), and not guaranteed to resolve to the right branch (if a commit is shared by multiple branches) by git remote transfer protocols.

---

Maybe you are right. But "master" is also wrong for repositories which use "dev", "main" or some other name. What would you propose?

---

I'll make an additional pull request without that commit. As I always say: don't mix too many commits in the same pull request.

---

> Maybe you are right. But "master" is also wrong for repositories which use "dev", "main" or some other name. What would you propose?

Yes, we've had ocrd_cis:dev for a while – it was an extra hickup for @kba each time as I recall. If indeed more and more repos play this game, then we need something robust that does work under all circumstances (or at least fails loudly).

Perhaps someone else with more git fu?

---

> I'll make an additional pull request without that commit. As I always say: don't mix too many commits in the same pull request.

Please don't – these belong together.

---

The reason for splitting this in two PRs is quite simple: the first commit fixes things for everybody, the second fix (which obviously needs more thoughts) is required only for a few people like @kba.

Therefore I'd prefer to have the first commit (in PR #276) merged soon.